docs/config/ca-certificate-bundle-for-imagestream-backups.md

```diff
@@ -0,0 +1,959 @@
+   - DPA is the source of truth for CA certificate bundle
+
+3. **Certificate File Changes**: Not supported
+   - Changes directly to files on disk (bypassing ConfigMap)
+   - Not detected or monitored
+
+**Predicate Filtering** (from `predicate.go`):
+
+The controller uses `veleroPredicate()` to filter events:
+- **Update events**: Only trigger if `generation` changed (spec modification)
+- **Create events**: Trigger if resource has `openshift.io/oadp` label or is DPA
+- **Delete events**: Trigger if resource has `openshift.io/oadp` label or is DPA
+- This prevents status-only updates from triggering unnecessary reconciliations
+
+**Typical Reconciliation Scenarios**:
+
+1. **User edits DPA CA cert**: Instant → ConfigMap update → 1-2 min file sync → new backups use cert
+2. **User adds new BSL with CA**: Instant (owned resource) → ConfigMap update → 1-2 min → effective
+3. **User updates BSL credential secret**: Instant (if labeled) → full reconciliation → ConfigMap update
+4. **User manually creates BSL with CA outside DPA**: No trigger → discovered at next DPA reconciliation
+5. **Velero updates BSL status**: No trigger (generation unchanged, status-only update filtered by predicate)
+
+### When Velero Pod Restarts
+
+See [Pod Restart Triggers](#pod-restart-triggers) in Key Concepts for the complete list of conditions that cause pod restarts vs those that don't.
+
+**Impact on running backups**: Pod restarts cause running imagestream backups to fail. ConfigMap-only updates allow running backups to complete while new backups use updated certificates after the kubelet sync period.
+
+### Triggering Discovery of Non-DPA BSL Changes
+
+Non-OADP BSLs (BackupStorageLocations created outside of DPA spec) are discovered via `r.List()` call in `processCACertForBSLs()` during each reconciliation. They do NOT automatically trigger reconciliation when modified.
+
+**Safe trigger mechanisms** (ConfigMap-only update, no pod restart):
+
+1. **DPA Annotation (Recommended)**:
+   ```bash
+   oc annotate dpa <dpa-name> -n openshift-adp reconcile=$(date +%s)
+   ```
+   - Triggers immediate reconciliation
+   - Updates ConfigMap if certificates changed
+   - Does NOT modify deployment spec
+   - Does NOT restart Velero pod
+
+2. **DPA Metadata Update (Alternative)**:
+   ```bash
+   oc patch dpa <dpa-name> -n openshift-adp --type=merge -p '{"metadata":{"labels":{"last-sync":"'$(date +%s)'"}}}'
+   ```
+   - Triggers reconciliation via metadata change
+   - Safe - metadata changes don't affect deployment
+
+**Unsafe mechanisms** (causes pod restart):
+- ❌ Toggling `backupImages` setting
+- ❌ Adding/removing DPA `spec.backupLocations` unnecessarily
+
+**Future improvement**: OADP may implement a watch on all BSLs in the namespace (not just owned ones) to automatically detect Non-DPA BSL changes, eliminating the need for manual triggering. Currently, `Owns(&velerov1.BackupStorageLocation{})` only watches OADP-created BSLs.
```

Comment:
This is an enhancement that we can do now as immediate follow up if necessary.. or later follow up.